classes/hooks.php

```diff
@@ -149,25 +159,35 @@ public function check_order_received() {
 	 */
 	public function handle_refund_callback() {
 		try {
-			$refund = Plugin::gateway()->receive_refund_callback();
-			$meta   = $refund->getMerchantMetadata();
+			$refund     = Plugin::gateway()->receive_refund_callback();
+			$meta       = $refund->getMerchantMetadata();
+			$payment_id = $refund->getPaymentId();
 
 			ZCO()->logger()->debug( 'Received Zaver refund callback', (array) $refund );
 
-			if ( ! isset( $meta['orderId'] ) ) {
+			$order_id = Helper::get_order_by_payment_id( $payment_id );
+			if ( empty( $order_id ) ) {
+				$order_id = $meta['orderId'] ?? null;
+			}
```

Comment:
Question: If we should have a orderId here, why not use that over the meta data lookup first? It will be a lot cheaper to do that rather then searching for the metadata. Or is there a reason we do it in this order instead?

Reply:
This is related to the initial issue: multiple payment ID related to the same orderId which is exactly what we want to avoid here. Every time the customers clicks on the place order button, a new payment ID is created. If they go back to the WC store without changing the cart, a new payment ID will be created again. We're only interested in the one that was actually saved to the order. As for Zaver, they don't seem to distinguish between an actually finalized order, and an aborted; they will keep sending callbacks to the store with the aborted payment ID(s).